homeassistant/components/homewizard/sensor.py

```diff
@@ -106,6 +161,122 @@
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="active_current_l3_a",
+        name="Active current L3",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="active_frequency_hz",
+        name="Active frequency",
+        native_unit_of_measurement=UnitOfFrequency.HERTZ,
+        device_class=SensorDeviceClass.FREQUENCY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="voltage_sag_l1_count",
+        name="Voltage sags detected L1",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="voltage_sag_l2_count",
+        name="Voltage sags detected L2",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="voltage_sag_l3_count",
+        name="Voltage sags detected L3",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="voltage_swell_l1_count",
+        name="Voltage swells detected L1",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="voltage_swell_l2_count",
+        name="Voltage swells detected L2",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="voltage_swell_l3_count",
+        name="Voltage swells detected L3",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="any_power_fail_count",
+        name="Power failures detected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="long_power_fail_count",
+        name="Long power failures detected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="active_power_average_w",
+        name="Active average demand",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="montly_power_peak_w",
+        name="Peak demand current month",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key="montly_power_peak_timestamp",
+        name="Peak demand detected timestamp",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorDeviceClass.TIMESTAMP,
+    ),
```

Comment:
I'm a bit unsure about this one. We removed 'gas timestamp' as it is automatically set by HA as last update when 'Total gas' value is updated. This value is important for Belgium users to know when the net supplier detected a peak.

Should we add this? Here or as attribute?

Reply:
Removed and planned to be added as 'attribute'.